includes/class-wc-payment-gateway-wcpay.php

```diff
@@ -1685,6 +1685,23 @@ public function capture_charge( $order ) {
 		$currency                 = WC_Payments_Utils::get_order_intent_currency( $order );
 
 		try {
+			$intent_id    = $order->get_transaction_id();
+			$intent       = $this->payments_api_client->get_intent( $intent_id );
+			$payment_type = $this->is_payment_recurring( $order->get_id() ) ? Payment_Type::RECURRING() : Payment_Type::SINGLE();
+
+			$metadata_from_intent = $intent->get_metadata(); // mobile app may have set metadata.
+			$metadata_from_order  = $this->get_metadata_from_order( $order, $payment_type );
+			$merged_metadata      = array_merge( $metadata_from_order, $metadata_from_intent ); // prioritize metadata from mobile app.
+
+			$this->payments_api_client->update_intention_unrestricted(
+				$intent_id,
+				null,
+				null,
+				null,
+				null,
+				$merged_metadata
+			);
+
```

Comment:
I wonder if it would be possible to extract this into a separate method, say in the `Order` class. This code flow requires `$order` as an input and updates related intent, so maybe having something like `$order->refresh_intention_metadata` would make it look cleaner in this method and allow for re-use whenever needed.